analyzers/src/SonarAnalyzer.VisualBasic/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -89,6 +89,7 @@ static bool TakesExpressionTree(SymbolInfo info)
                 EnumMemberDeclarationSyntax x => x.Identifier,
                 InvocationExpressionSyntax x => x.Expression?.GetIdentifier(),
                 ModifiedIdentifierSyntax x => x.Identifier,
+                ObjectCreationExpressionSyntax x => x.Type?.GetIdentifier(),
```

Comment:
I do not understand the "remove unsused API sentiment". This is an internal API and the UTs use it. The API should be complete. Otherwise, we end up in a situation like the registrations where we have a support matrix nobody understands and, therefore, the API is used incorrectly or untrusted.
An API should be either complete or throw in unsupported scenarios. But silently behaving wrongly or being incomplete is bad practice.